exporters/otlp/otlplog/otlploghttp/config.go

```diff
@@ -193,6 +267,53 @@ type HTTPTransportProxyFunc func(*http.Request) (*url.URL, error)
 // proxy to use for an HTTP request. If this option is not used, the client
 // will use [http.ProxyFromEnvironment].
 func WithProxy(pf HTTPTransportProxyFunc) Option {
-	// TODO: implement.
-	return nil
+	return fnOpt(func(c config) config {
+		c.proxy = newSetting(pf)
+		return c
+	})
+}
+
+// setting is a configuration setting value.
+type setting[T any] struct {
+	Value T
+	Set   bool
+}
+
+// newSetting returns a new [setting] with the value set.
```

Comment:
```suggestion
// newSetting returns a new setting with the value set.
```

It is unexported so I think it is worth to add `[]`